src/ServiceControlInstaller.Engine/Instances/ServiceControlInstance.cs

```diff
@@ -344,8 +344,17 @@ public bool TryStopService()
                 return true;
             }
 
-            Service.Stop();
+            try
+            {
 
+                // Will throw with "The service cannot accept control messages at this time" message if the service is in a pending state 
+                Service.Stop();
+            }
+            catch (InvalidOperationException)
```

Comment:
Can any other than "The service cannot accept control messages" InvalidOperationException happen here? i.e. do we risk swallowing something we should not?